WooCommerce/Classes/Analytics/WooAnalyticsEvent.swift

```diff
@@ -535,3 +537,203 @@ extension WooAnalyticsEvent {
         WooAnalyticsEvent(statName: .jetpackInstallButtonTapped, properties: ["source": source.rawValue])
     }
 }
+
+// MARK: - In Person Payments
+//
+extension WooAnalyticsEvent {
+
+    enum InPersonPayments {
+
+        enum Keys {
+            static let batteryLevel = "battery_level"
+            static let gatewayID = "plugin_slug"
+            static let errorDescription = "error_description"
+            static let softwareUpdateType = "software_update_type"
+        }
+
+        static let unknownGatewayID = "unknown"
```

Comment:
I agree. A nil should not be possible, but since we have distributed stores, it is only avoided right now by temporal dependencies enforced by our view progression or states in CardReaderConnectionController.

I'd like to leave it in for safety and we could use it as an indicator that something is wrong.

Reply:
We use "unknown" pluginSlug for all the `CARD_PRESENT_ONBOARDING_NOT_COMPLETED` events which are tracked before the app knows which plugins are installed. We could not track the plugin slug in those cases 🤷 . Either way as discussed on Slack I'd suggest handling the `CARD_PRESENT_ONBOARDING_NOT_COMPLETED` after the code freeze as it's not an urgent issue.